libevent-sys/build.rs

```diff
@@ -1,18 +1,163 @@
+    println!("cargo:rustc-link-lib=static=event_extra");
+
+    if cfg!(feature = "openssl") || cfg!(feature = "openssl_bundled") {
+        println!("cargo:rustc-link-lib=static=event_openssl");
+    }
+
+    if cfg!(feature = "threading") {
+        println!("cargo:rustc-link-lib=static=event_pthreads");
+    }
+
+    println!("cargo:include={}/include", dst.display());
+
+    dst
+}
+
+#[cfg(not(feature = "bundled"))]
+fn run_pkg_config() -> Option<Vec<String>> {
+    let mut pkg = pkg_config::Config::new();
+    pkg.cargo_metadata(false)
         .atleast_version("2")
-        .statik(cfg!(feature = "static"))
-        .probe("libevent")
-        .unwrap();
+        .statik(cfg!(feature = "static"));
+
+    let mut lib = match pkg.probe("libevent_core") {
+        Ok(lib) => lib,
+        Err(_e) => { return None; }
+    };
+    // TODO: Probably combine all pc includes, just to be safe.
+
+    pkg.cargo_metadata(true).probe("libevent_core").unwrap();
+    pkg.cargo_metadata(true).probe("libevent_extra").unwrap();
+
+    if cfg!(feature = "openssl") {
+        pkg.cargo_metadata(true).probe("libevent_openssl").unwrap();
+    }
+
+    if cfg!(feature = "threading") {
+        pkg.cargo_metadata(true).probe("libevent_pthreads").unwrap();
+    }
+
+    let include_paths = lib.include_paths.drain(..).map(|path| {
+        let path_s = path.to_str().unwrap().to_string();
+        println!("cargo:include={}", &path_s);
+        path_s
+    }).collect();
+
+    Some(include_paths)
+}
+
+#[cfg(feature = "bundled")]
+fn find_libevent() -> Option<Vec<String>> {
+    use std::path::Path;
+    use std::process::Command;
+
+    if !Path::new("libevent/.git").exists() {
+        Command::new("git").args(&["submodule", "update", "--init"])
+            .status().expect("Running `git submodule init` failed.");
+    } else {
+        Command::new("git").args(&["submodule", "update", "--recursive"])
+            .status().expect("Running `git submodule update` failed.");
+    }
+
+    Some(vec![format!("{}/include", build_libevent("libevent").display())])
+}
+#[cfg(not(feature = "bundled"))]
+fn find_libevent() -> Option<Vec<String>> {
+    run_pkg_config()
+}
+
+fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=libevent");
+    println!("cargo:rerun-if-changed=wrapper.h");
+
+    if cfg!(feature = "verbose_build") {
+        for (key, val) in env::vars() {
+            println!("{}: {}", key, val);
+        }
+        let args: Vec<String> = env::args().collect();
+        println!("args: {:?}", args);
+    }
+
+    let include_paths = find_libevent()
+        .expect("No include paths for libevent found");
+
+    let mut builder = bindgen::Builder::default();
+
+    // For some reason, bindgen fails to find the paths for cross-compilation
+    // builds. Just add each path since we have the info anyway.
+    for path in include_paths {
+        builder = builder.clang_arg(format!("-I{}", path));
```

Comment:
This seems a bit curious.  Did you find anything on why this is happening?

Did you also try `detect_include_paths`? https://docs.rs/bindgen/0.53.2/bindgen/struct.Builder.html#method.detect_include_paths

Reply:
Wishful thinking on my part that bindgen would somehow detect the emitted `cargo:include` metadata, which is only useful for projects wanting to use the `DEP_` variables explicitly. I also see that many crates inject the `-target` flag into clang when cross-compiling (like [sdl2-sys](https://github.com/Rust-SDL2/rust-sdl2/blob/3c6e104ec89fb5cce193ec4e11d385a5df342d88/sdl2-sys/build.rs#L581)), so I should do that as well.